src/Npgsql/TypeHandlers/JsonbHandler.cs

```diff
@@ -77,6 +77,9 @@ public override int ValidateAndGetLength(char[] value, ref NpgsqlLengthCache len
             return base.ValidateAndGetLength(value, ref lengthCache, parameter) + 1;
         }
 
+        internal override int ValidateAndGetLengthBytes(byte[] value)
```

Comment:
Is there any reason we need this additional virtual method, as opposed to putting the logic directly in `ValidateAndGetLength(byte[])`?

It's true that an explicit interface implementation cannot be virtual and overridden, but you can simply have JsonHandler implement `INpgsqlTypeHandler<byte[]>` again and provide its own explicit interface implementation for `ValidateAndGetLength(byte[])`.

Reply:
See discussion on #1891. It was suggested that I do it this way by @YohDeadfall.

Reply:
Ah yes, I can see, sorry again for joining a bit late.

Unless I'm mistaken, both TextHandler and Jsonbhandler can implement `INpgsqlTypeHandler<byte[]>`, each providing their own explicit interface implementation which will get called. You can give this a try, it should cut down on some code.

Reply:
@roji Then it wouldn't be possible to call base methods. Currently `TextHandler` processes bytes, and `JsonbHandler` just adjusts the logic to use `JsonbProtocolVersion`.

Reply:
Ah yes, you're right - calling the base class's implementation isn't possible, thanks.